relayer/crates/starknet-chain-components/src/impls/queries/client_state.rs

```diff
@@ -84,27 +91,54 @@ where
 impl<Chain, Counterparty> ClientStateWithProofsQuerier<Chain, Counterparty>
     for QueryCometClientState
 where
-    Chain: HasClientIdType<Counterparty>
+    Chain: HasClientIdType<Counterparty, ClientId = ClientId>
         + HasHeightType<Height = u64>
+        + CanQueryChainStatus<ChainStatus = StarknetChainStatus>
+        + HasIbcCommitmentPrefix<CommitmentPrefix = Vec<u8>>
         + HasCommitmentProofType<CommitmentProof = StarknetCommitmentProof>
         + CanQueryClientState<Counterparty>
+        + HasStarknetProofSigner<ProofSigner = Secp256k1KeyPair>
+        + CanRaiseAsyncError<String>
         + HasAsyncErrorType,
-    Counterparty: HasClientStateType<Chain> + HasHeightType,
+    Counterparty: HasClientStateType<Chain, ClientState = CometClientState> + HasHeightType,
 {
     async fn query_client_state_with_proofs(
         chain: &Chain,
         tag: PhantomData<Counterparty>,
         client_id: &Chain::ClientId,
         query_height: &Chain::Height,
     ) -> Result<(Counterparty::ClientState, Chain::CommitmentProof), Chain::Error> {
-        // FIXME: properly fetch consensus state with proofs
+        // FIXME: properly fetch client state with proofs
         let client_state = chain
             .query_client_state(tag, client_id, query_height)
             .await?;
 
+        let chain_status = chain.query_chain_status().await?;
+
+        // FIXME: CometClientState can't be encoded to protobuf
+        let protobuf_encoded_client_state = Vec::new();
```

Comment:
do we have an implementation that encodes `CometClientState` to protobuf encoding?

cc @soareschen

PS. it's not important for demo2, as ibc-go currently ignored self client/consensus validation.

Reply:
We don't have protobuf encoding for `CometClientState`. But what we should probably do is to use back `TendermintClientState` and implement Cairo encoding for that. We currently have `CometClientState` because the fields don't exactly match.